cmd/main.go

```diff
@@ -178,15 +194,21 @@ func setupManager(flagVar *flags.FlagVar, cacheOptions cache.Options, scheme *ma
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
-	if flagVar.DropStoredVersion != "" {
-		go func(version string) {
-			dropStoredVersion(mgr, version)
-		}(flagVar.DropStoredVersion)
-	}
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+}
+
+func runKymaMetricsCleanup(kymaMetrics *metrics.KymaMetrics, kcpClient client.Client,
+	cleanupIntervalInMinutes time.Duration) {
+	scheduler := gocron.NewScheduler(time.UTC)
+	_, scheduleErr := scheduler.Every(cleanupIntervalInMinutes).Minutes().Do(func() {
+		if err := kymaMetrics.CleanupNonExistingKymaCrsMetrics(context.TODO(), kcpClient); err != nil {
```

Comment:
After reading a bit about context:
- https://medium.com/@jamal.kaksouri/the-complete-guide-to-context-in-golang-efficient-concurrency-management-43d722f6eaea
- https://stackoverflow.com/questions/74239074/context-todo-or-context-background-which-one-should-i-prefer

Should we rather use `context.Background()` here? My understanding is that TODO would indicate that we replace it with some parent context at some point, but I don't think this will be the case here?

Reply:
yes I actually had this just as a placeholder while writing the function and was just looking into it :D